lib/mqtt.ts

```diff
@@ -66,18 +71,123 @@ export interface MqttSubscribeRequest extends MqttRequest {
 
 type Payload = string | Object | DataView;
 
-export class Connection {
+export class AwsIotMqttConnectionConfigBuilder {
+    private params: ConnectionConfig   
+    private tls_ctx_options?: io.TlsContextOptions
+
+    private constructor() {
+        this.params = {
+            client_id: '', 
+            host_name: '',
+            connect_timeout: 3000, 
+            port: io.is_alpn_available() ? 443 : 8883,
+            use_websocket: false,
+            clean_session: false,
+            keep_alive: undefined,
+            will: undefined,
+            username: '?SDK=NodeJSv2&Version=0.2.0',
+            password: undefined,
+            tls_ctx: undefined,
+        };
+    }
+
+    static new_mtls_builder_from_path(cert_path: string, key_path: string) {
+        let builder = new AwsIotMqttConnectionConfigBuilder();
+        builder.tls_ctx_options = io.TlsContextOptions.create_client_with_mtls(cert_path, key_path);
+        
+        if (io.is_alpn_available()) {
+            builder.tls_ctx_options.alpn_list = 'x-amzn-mqtt-ca';
+        }   
+        
+        return builder;
+    }  
+
+    with_certificate_authority_from_path(ca_path?: string, ca_file?: string) {
+        if (this.tls_ctx_options !== undefined) {
+            this.tls_ctx_options.override_default_trust_store(ca_path, ca_file);
+        }
+
+        return this;
+    }
+
+    with_endpoint(endpoint: string) {
+        this.params.host_name = endpoint;
+        return this;
+    }
+
+    with_client_id(client_id: string) {
+        this.params.client_id = client_id;
+        return this;
+    }
+
+    with_clean_session(clean_session: boolean) {
+        this.params.clean_session = clean_session;
+        return this;
+    }
+
+    with_use_websockets() {
+        this.params.use_websocket = true;
+
+        if (this.tls_ctx_options !== undefined) {
+            this.tls_ctx_options.alpn_list = undefined;
+            this.params.port = 443;
+        }
+
+        return this;
+    }
+
+    with_keep_alive_seconds(keep_alive: number) {
+        this.params.keep_alive = keep_alive;
+        return this;
+    }
+
+    with_timeout_ms(timeout_ms: number) {
+        this.params.timeout = timeout_ms;
+        return this;
+    }
+
+    with_will(will: string) {
+        this.params.will = will;
+        return this;
+    }
+
+    with_connect_timeout_ms(timeout: number) {
+        this.params.connect_timeout = timeout;
+        return this;
+    }
+
+    build() {
+        if (this.params.client_id === undefined || this.params.host_name === undefined) {
+            throw 'client_id and endpoint are required';
+        }
+
+        if (this.tls_ctx_options === undefined) {
```

Comment:
The other CRTs do not require TLS for all connections, mostly for testing.

Reply:
this is the aws iot specific one. You can create a config object manually and use that if you want.